test/e2e/performanceprofile/functests/utils/nodepools/nodepools.go

```diff
@@ -44,3 +45,20 @@ func waitForCondition(ctx context.Context, c client.Client, NpName, namespace st
 		return conditionFunc(np.Status.Conditions), nil
 	})
 }
+
+func GetByClusterName(ctx context.Context, c client.Client, hostedClusterName string) (*hypershiftv1beta1.NodePool, error) {
+	npList := &hypershiftv1beta1.NodePoolList{}
+	if err := c.List(ctx, npList); err != nil {
+		return nil, err
+	}
+	var np *hypershiftv1beta1.NodePool
+	for i := 0; i < len(npList.Items); i++ {
+		if npList.Items[i].Spec.ClusterName == hostedClusterName {
+			np = &npList.Items[i]
```

Comment:
break or return here, likely better to break

Reply:
Done.